src/reducers/versions.spec.tsx

```diff
@@ -616,4 +619,237 @@ describe(__filename, () => {
       });
     });
   });
+
+  describe('createInternalDiffs', () => {
+    type CreateInternalDiffsParams = {
+      baseVersionId?: number;
+      headVersionId?: number;
+      version: ExternalVersionWithDiff;
+    };
+
+    const _createInternalDiffs = ({
+      baseVersionId = 1,
+      headVersionId = 2,
+      version,
+    }: CreateInternalDiffsParams) => {
+      return createInternalDiffs({ baseVersionId, headVersionId, version });
+    };
+
+    const createVersionWithChange = (change = {}) => {
+      return {
+        ...fakeVersion,
+        file: {
+          ...fakeVersion.file,
+          diff: [
+            {
+              ...fakeExternalDiff,
+              hunks: [
+                {
+                  ...fakeExternalDiff.hunks[0],
+                  changes: [
+                    {
+                      ...fakeExternalDiff.hunks[0].changes[0],
+                      ...change,
+                    },
+                  ],
+                },
+              ],
+            },
+          ],
+        },
+      };
+    };
+
+    it('creates an array of DiffInfo objects from a version with diff', () => {
+      const baseVersionId = 132;
+      const headVersionId = 133;
+      const externalDiffs = [fakeExternalDiff];
+      const version = {
+        ...fakeVersion,
+        file: {
+          ...fakeVersion.file,
+          diff: externalDiffs,
+        },
+      };
+
+      const diffs = _createInternalDiffs({
+        baseVersionId,
+        headVersionId,
+        version,
+      });
+
+      expect(diffs).toHaveLength(externalDiffs.length);
+      diffs.forEach((diff, index) => {
+        const externalDiff = externalDiffs[index];
+
+        expect(diff).toHaveProperty('oldRevision', String(baseVersionId));
+        expect(diff).toHaveProperty('newRevision', String(headVersionId));
+        expect(diff).toHaveProperty('oldMode', externalDiff.mode);
+        expect(diff).toHaveProperty('newMode', externalDiff.mode);
+        expect(diff).toHaveProperty('oldPath', externalDiff.old_path);
+        expect(diff).toHaveProperty('newPath', externalDiff.path);
+        expect(diff).toHaveProperty(
+          'oldEndingNewLine',
+          externalDiff.old_ending_new_line,
+        );
+        expect(diff).toHaveProperty(
+          'newEndingNewLine',
+          externalDiff.new_ending_new_line,
+        );
+
+        // These props will be tested later.
```

Comment:
Nit: When I first encountered this comment I thought you meant this was being deferred to a follow-up, so maybe saying something like "These props will be tested in a different test case below." would make it a bit clearer?

Reply:
:+1: